packages/core/admin/admin/src/pages/Users/ListPage/DynamicTable/TableHead/index.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {
+  BaseCheckbox,
+  IconButton,
+  TableLabel,
+  Th,
+  Thead,
+  Tr,
+  Tooltip,
+  VisuallyHidden,
+} from '@strapi/parts';
+import { SortIcon, useQueryParams } from '@strapi/helper-plugin';
+import PropTypes from 'prop-types';
+
+const TableHead = ({
+  areAllEntriesSelected,
+  entriesToDelete,
+  headers,
+  onSelectAll,
+  withMainAction,
+  withBulkActions,
+}) => {
+  const [{ query }, setQuery] = useQueryParams();
+  const sort = query.sort || '';
+  const [sortBy, sortOrder] = sort.split(':');
+
+  const isIndeterminate = !areAllEntriesSelected && entriesToDelete.length;
+
+  return (
+    <Thead>
+      <Tr>
+        {withMainAction && (
+          <Th>
+            <BaseCheckbox
+              aria-label="Select all entries"
+              checked={areAllEntriesSelected}
+              indeterminate={isIndeterminate}
+              onChange={onSelectAll}
+            />
+          </Th>
+        )}
+        {headers.map(({ name, metadatas: { sortable: isSortable, label } }) => {
+          const isSorted = sortBy === name;
+          const isUp = sortOrder === 'ASC';
+
+          const handleClickSort = (shouldAllowClick = true) => {
+            if (isSortable && shouldAllowClick) {
+              const nextSortOrder = isSorted && sortOrder === 'ASC' ? 'DESC' : 'ASC';
+              const nextSort = `${name}:${nextSortOrder}`;
+
+              setQuery({
+                sort: nextSort,
+              });
+            }
+          };
+
+          return (
+            <Th
+              key={name}
+              action={
+                isSorted ? (
+                  <IconButton
+                    label={`Sort on ${label}`}
+                    onClick={handleClickSort}
+                    icon={isSorted ? <SortIcon isUp={isUp} /> : undefined}
+                    noBorder
+                  />
+                ) : (
+                  undefined
+                )
+              }
+            >
+              <Tooltip label={`Sort on ${label}`}>
+                <TableLabel
+                  as={!isSorted && isSortable ? 'button' : 'span'}
+                  label={`Sort on ${label}`}
+                  onClick={() => handleClickSort(!isSorted)}
```

Comment:
What do you guys think about the sort feature having a better loading state, maybe reloading only the table rows ? Now it's loading on the whole page. Ideally an overlay on the table would be better. Or we can talk to maeva about this one. 

Because it's affecting the a11y : when the sorting is applied we loose focus on everything since the div is disappearing.

Reply:
We will need a design for this right now the table does not have a loading state, we might improve this later.

Reply:
I did something in this [PR](https://github.com/strapi/strapi/pull/10869) that fixes the issue